service/resource_router.go

```diff
@@ -0,0 +1,68 @@
+package service
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+
+	"github.com/giantswarm/aws-operator/service/key"
+)
+
+const (
+	legacyResourceName = "legacy"
+)
+
+// NewResourceRouter determines which resources are enabled based upon the
+// version in the version bundle.
+func NewResourceRouter(rs []framework.Resource) func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
```

Comment:
I was wondering why this takes a list of resources. It looks like below is some filtering applied on all existing resources. I think it would be good to define the list of resources being used explicitly. It would be safer and better to provide a map of resource lists and based on the versions we just take the list from the map `map[string][]framework.Resource`, where `string` is a version bundle version. The benefit would be that we do not need to compute the same lists at runtime over and over again and we cannot get trapped by any eventual flaw in the filtering. We can just safe us from this hustle. The router then becomes leaner and the initial resource list/map creation more straight forward and deterministic.

Reply:
Yes that makes total sense. I've removed the filtering and the router now takes in the map.